packages/mobile-app/data/facades/chain/oreowalletHandlers.ts

```diff
@@ -0,0 +1,51 @@
+import { f } from "data-facade";
+import { Asset, ChainHandlers } from "./types";
+
+import { isValidPublicAddress } from "ironfish-native-module";
+import { Network } from "../../constants";
+import { oreoWallet } from "../../wallet/oreowalletWallet";
+
+export const chainHandlers = f.facade<ChainHandlers>({
+  getAsset: f.handler.query(
+    async ({ assetId }: { assetId: string }): Promise<Asset | null> => {
+      const asset = await oreoWallet.getAsset(Network.MAINNET, assetId);
+
+      if (!asset) {
+        return null;
+      }
+
+      return {
+        id: assetId,
+        name: asset.name,
+        createdTransactionHash: asset.createdTransactionHash,
+        createdTransactionTimestamp: asset.createdTransactionTimestamp,
+        creator: asset.creator,
+        metadata: asset.metadata,
+        owner: asset.owner,
+        verification: !asset.verified
+          ? { status: "unverified" }
+          : {
+              status: "verified",
+              symbol: asset.symbol ?? asset.name,
+              decimals: asset.decimals ?? undefined,
+              logoURI: asset.logoURI ?? undefined,
+              website: asset.website ?? undefined,
+            },
+        supply: asset.supply,
+      };
+    },
+  ),
+  getNetworkInfo: f.handler.query(async () => {
+    // TODO: Implement network switching
+    return { networkId: 0 };
+  }),
+  isValidPublicAddress: f.handler.query(({ address }: { address: string }) => {
+    return isValidPublicAddress(address);
```

Comment:
suggestion: Can we call this directly instead of going through the data facade?

Reply:
I'd prefer to keep business logic separated from the React code in the facade+wallet layer, and it also mirrors how we do it in the node app (though I know there it's not really an option to call it in the React code). I think the React implementation would basically be a react-query query with this same function anyway?

Reply:
Ah, is it async? If so, then I agree keeping it here makes more sense